vault/expiration.go

```diff
@@ -82,7 +82,7 @@ type ExpirationManager struct {
 	restoreLoaded      sync.Map
 	quitCh             chan struct{}
 
-	coreStateLock     *sync.RWMutex
+	coreStateLock     *DeadlockMutexRW
```

Comment:
Could you add a comment here about the buildtag-conditional behavior, as well as what one should expect if the deadlock detection is enabled?

Reply:
Good call, but I think I'd rather put the comment next to the type definition(s), since we may wind up using this in other places than just stateLock.  Would that be acceptable?